lib/Target/SubstraitPB/Export.cpp

```diff
@@ -18,23 +18,28 @@
 #include "llvm/ADT/TypeSwitch.h"
 #include "llvm/Support/ErrorHandling.h"
 
+// TODO(ingomueller): Find a way to make `substrait-cpp` declare these headers
+// as system headers and remove the diagnostic fiddling here.
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Weverything"
 #include <google/protobuf/text_format.h>
 #include <google/protobuf/util/json_util.h>
 #include <substrait/proto/algebra.pb.h>
 #include <substrait/proto/extensions/extensions.pb.h>
 #include <substrait/proto/plan.pb.h>
 #include <substrait/proto/type.pb.h>
+#pragma clang diagnostic pop
 
 using namespace mlir;
 using namespace mlir::substrait;
 using namespace mlir::substrait::protobuf_utils;
 using namespace ::substrait;
 using namespace ::substrait::proto;
 
-namespace pb = google::protobuf;
-
 namespace {
 
+namespace pb = ::google::protobuf;
+
```

Comment:
This nesting of namespaces might also have solved some windows issues... We'll see later today!